org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/handler/ThreadsRequestHandler.java

```diff
@@ -0,0 +1,149 @@
+/*******************************************************************************
+* Copyright (c) 2017 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License v1.0
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v10.html
+*
+* Contributors:
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.jdt.ls.debug.adapter.handler;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.jdt.ls.debug.DebugUtility;
+import org.eclipse.jdt.ls.debug.adapter.AdapterUtils;
+import org.eclipse.jdt.ls.debug.adapter.ErrorCode;
+import org.eclipse.jdt.ls.debug.adapter.Events;
+import org.eclipse.jdt.ls.debug.adapter.IDebugAdapterContext;
+import org.eclipse.jdt.ls.debug.adapter.IDebugRequestHandler;
+import org.eclipse.jdt.ls.debug.adapter.Messages.Response;
+import org.eclipse.jdt.ls.debug.adapter.Requests;
+import org.eclipse.jdt.ls.debug.adapter.Requests.Arguments;
+import org.eclipse.jdt.ls.debug.adapter.Requests.Command;
+import org.eclipse.jdt.ls.debug.adapter.Requests.ContinueArguments;
+import org.eclipse.jdt.ls.debug.adapter.Requests.NextArguments;
+import org.eclipse.jdt.ls.debug.adapter.Requests.PauseArguments;
+import org.eclipse.jdt.ls.debug.adapter.Requests.StepInArguments;
+import org.eclipse.jdt.ls.debug.adapter.Requests.StepOutArguments;
+import org.eclipse.jdt.ls.debug.adapter.Requests.ThreadsArguments;
+import org.eclipse.jdt.ls.debug.adapter.Responses;
+import org.eclipse.jdt.ls.debug.adapter.Types;
+
+import com.sun.jdi.ThreadReference;
+import com.sun.jdi.VMDisconnectedException;
+
+public class ThreadsRequestHandler implements IDebugRequestHandler {

```

Comment:
single responsibility per class?

Reply:
When your PR #70 is ready, i will split it to separate handler later.